src/formatDate.js

```diff
@@ -8,7 +8,27 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
+
+  const dateParts = date.split(separatorFrom);
+  const dateMap = {};
+
+  fromFormat.forEach((part, index) => {
+    dateMap[part] = dateParts[index];
+  });
+
+  const formattedDateParts = toFormat.map((part) => {
+    if (part === 'YYYY' && dateMap['YY']) {
+      return dateMap['YY'] < 30 ? '20' + dateMap['YY'] : '19' + dateMap['YY'];
```

Comment:
The logic here assumes that if 'YYYY' is requested and 'YY' is provided, it will convert 'YY' to 'YYYY' based on a threshold of 30. This logic might not work correctly if 'YY' is not part of the `fromFormat`. Ensure that 'YY' exists in `dateMap` before applying this logic.